### PR TITLE
Add simple test suite

### PR DESCRIPTION
**Pull request by wilzbach**
With the improvement ideas from #15

### CODE REVIEW

File: source/app.d [hunk 1]
```diff
@@ -70,13 +98,21 @@ void githubHook(HTTPServerRequest req, HTTPServerResponse res)
...
         return res.writeBody("ignored");
     }
 }
 
+auto runTaskHelper(Fun, Args...)(Fun fun, Args args)
+{
+    if (runAsync)
+        runTask(fun, args);
+    else
+        return fun(args);
```
Comment (wilzbach) [inline source/app.d]: > Actually the async processing isn't really safe and might loose jobs, maybe we never want to do things asynchronously?

I don't know how "safe" Vibe.d's `runTask` is, so leaving this decision to you.

File: test/utils.d [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    else
+    {
+        assert(0, "Request for unexpected URL received: " ~ req.requestURL);
+    }
+
+    if (expectation.reqHandler !is null)
+        return expectation.reqHandler(req, res);
+
+    string filePath = buildPath(payloadDir, req.requestURL[1 .. $].replace("/", "_"));
```
Comment (wilzbach) [inline test/utils.d]: I moved all payloads in one folder and use this `/` -> `_` replacement rule, s.t. one can have a good overview of all available payloads. 
It's still simple to fetch new requests, e.g.

```
api=repos/dlang/dmd/pulls/6328/commits && curl https://api.github.com/${api} > data/payloads/github_$(echo $api | tr '/' '_')
```

However if you prefer to have an exact 1:1 matching with folders, I wouldn't mind either.

Comment (MartinNowak) [inline test/utils.d]: Sounds fine for now.

File: test/comments.d [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+// existing dlang bot comment -> update comment
+unittest
+{
+    setAPIExpectations(
+        "/github/repos/dlang/phobos/pulls/4921/commits",
+        "/github/repos/dlang/phobos/issues/4921/comments",
+        "/bugzilla/buglist.cgi?bug_id=8573&ctype=csv&columnlist=short_desc",
+        "/github/repos/dlang/phobos/issues/comments/262784442",
+        (scope HTTPServerRequest req, scope HTTPServerResponse res){
```
Comment (MartinNowak) [inline test/comments.d]: That's fine and also comes with clearer error messages.

Technically it's possible to pass a polymorphic lambda to a template and check whether it's instantiatable though.
The error messages are not so good, b/c you can't know which type was meant.

```d
void foo(Args...)()
{
    static if (__traits(compiles, Args[0](12)))
    {
        pragma(msg, "int");
    }
    else static if (__traits(compiles, Args[0]("string")))
    {
        pragma(msg, "string");
    }
}

void test()
{
    foo!(a => 2 * a);
    foo!(a => a ~ "foo");
}
```d

**Pull request merged**